CMakeLists.txt

```diff
@@ -50,18 +52,28 @@ file(GLOB SOURCES "edlib/src/*.cpp")
 
 # Create libraries.
 add_library(edlib ${SOURCES})
+add_library(edlib::edlib ALIAS edlib)
 set_target_properties(edlib
-    PROPERTIES
+  PROPERTIES
+    CXX_VISIBILITY_PRESET hidden
+    VISIBILITY_INLINES_HIDDEN ON
     VERSION ${PROJECT_VERSION}
-    SOVERSION ${PROJECT_VERSION})
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+if(BUILD_SHARED_LIBS)
```

Comment:
Why do we have this under the flag, why not just build both static and dynamic library each time? If we are going to keep this flag, could you help me correctly passing it to CMake in .travis.yml and appveyor.yml, so that building of shared library is tested there?

Reply:
the problem being, in many cases consumers don't want to link against static libraries (distros hate this, because it makes tracking dependencies harder).

Reply:
@SoapZA aha so consumers want shared libraries, but distro package maintainers dont because they are harder to maintain?
So what about this flag, I still don't quite get its use case? Are you for keeping it, or not?

Reply:
Ah I think I get it -> the point is, by having these flags, people can choose what they want to build, and it is actually important to them. But why don't they just build all of it and then use what they need?

Reply:
If you allow consumers to build both libraries, you'll have to give a different name to static and import lib on Windows. Moreover the CMakeLists.txt shall be modified to allow shared and static in the same build, and give a different name for imported CMake targets and pkg-config files. Finally, it increases build time whild consumers may want either static or shared, not both.

Reply:
Aha ok! Thanks for explanation, I wasn't aware of these use cases, but if that is so, great, let's configure both so that either static or shared library is built.